server/src/main/java/keywhiz/service/daos/DaoModule.java

```diff
@@ -1,9 +1,48 @@
 package keywhiz.service.daos;
 
 import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import keywhiz.service.config.Readonly;
+import keywhiz.service.config.Readwrite;
 
 public class DaoModule extends AbstractModule {
   @Override protected void configure() {
     bind(ClientMapper.class).toProvider(ClientMapper::new);
   }
+
+  @Provides
+  @Singleton
+  @Readonly
+  GroupDAO readonlyGroupDAO(GroupDAO.GroupDAOFactory factory) {
+    return factory.readonly();
+  }
+
+  @Provides
+  @Singleton
+  @Readwrite
+  GroupDAO readwriteGroupDAO(GroupDAO.GroupDAOFactory factory) {
+    return factory.readwrite();
+  }
```

Comment:
These direct bindings allow for simplification and hide the factories from our code and tests.